docs/tutorials/hierarchical_flow.rst

```diff
@@ -0,0 +1,664 @@
+Hierarchical Flow
+=================
+
+This tutorial walks you through using SC's hierarchical flow to create a 
+system-on-chip (SoC), the ZeroSoC, a simple system-on-chip meant to
+demonstrate SiliconCompiler's end-to-end design capabilities.
+
+.. image:: _images/zerosoc.png
+
+ZeroSoC is based on a subset of the IP used in the open-source `OpenTitan SoC
+<https://github.com/lowrisc/opentitan>`_. It includes:
+
+* An `Ibex <https://github.com/lowrisc/ibex>`_ RV32I processor
+* A GPIO controller with 32 configurable I/O ports
+* A UART peripheral
+* RAM (2KB for ASIC, 8KB for FPGA)
+
+ZeroSoC's design and supporting files can be found at
+`<https://github.com/siliconcompiler/zerosoc>`_.
+
+Getting Started
+---------------
+Before we begin, the first thing to know is that the ZeroSoC ASIC is built in
+two parts:
+
+* First, the *core*, which contains the actual digital logic implementing the
+  ZeroSoC.
+* Second, the *top-level*, which wraps the core and places the ZeroSoC's padring
+  (the portion that implements I/O) around it.
+
+We build the ZeroSoC core and the top using SC's included :ref:`OpenROAD <openroad>`-based 
+:ref:`asicflow <asicflow-ref>`.
+For the *core*, we will use SC's 1) pin constraints (:keypath:`constraint,pin`)
+to define the location of the pins around the core, 2) component placement constraints
+(:keypath:`constraint,component`) to place the SRAM macro in the *core*, 3) supply a 
+power grid script to OpenROAD to build a power grid compatible with the hierarchal flow.
+For the *top-level*, we will reuse the *core* macro generated to demonstrate a 
+hierarchal flow.
+
+Let's begin by cloning the ZeroSoC Github repository. Be sure to initialize the
+submodules to pull down the required third-party design files:
+
+.. code-block:: console
+
+  $ git clone https://github.com/siliconcompiler/zerosoc.git
+  $ cd zerosoc
+  $ git submodule update --init --recursive
```

Comment:
Does this rely on the user to have set up sshkeys? It works when I have my sshkeys set up, but I don't, there's an error with cloning opentitan:
```
Cloning into '/home/auli4/zerosoc/opentitan'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:lowRISC/opentitan.git' into submodule path '/home/auli4/zerosoc/opentitan' failed
Failed to clone 'opentitan' a second time, aborting